system-helper/flatpak-system-helper.c

```diff
@@ -552,50 +549,14 @@ handle_deploy (FlatpakSystemHelper   *object,
           return G_DBUS_METHOD_INVOCATION_HANDLED;
         }
 
-      registry = flatpak_oci_registry_new (registry_uri, FALSE, -1, NULL, &error);
-      if (registry == NULL)
+      image_source = flatpak_image_source_new_local (registry_file, arg_ref, NULL, &error);
+      if (image_source == NULL)
         {
           g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
                                                  "Can't open child OCI registry: %s", error->message);
           return G_DBUS_METHOD_INVOCATION_HANDLED;
         }
 
-      index = flatpak_oci_registry_load_index (registry, NULL, &error);
-      if (index == NULL)
-        {
-          g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
-                                                 "Can't open child OCI registry index: %s", error->message);
-          return G_DBUS_METHOD_INVOCATION_HANDLED;
-        }
-
-      desc = flatpak_oci_index_get_manifest (index, arg_ref);
-      if (desc == NULL)
-        {
-          g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
-                                                 "Can't find ref %s in child OCI registry index", arg_ref);
-          return G_DBUS_METHOD_INVOCATION_HANDLED;
-        }
-
-      versioned = flatpak_oci_registry_load_versioned (registry, NULL, desc->parent.digest, (const char **)desc->parent.urls, NULL,
-                                                       NULL, &error);
-      if (versioned == NULL || !FLATPAK_IS_OCI_MANIFEST (versioned))
-        {
-          g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
-                                                 "Can't open child manifest");
-          return G_DBUS_METHOD_INVOCATION_HANDLED;
-        }
-
-      image_config = flatpak_oci_registry_load_image_config (registry, NULL,
-                                                             FLATPAK_OCI_MANIFEST (versioned)->config.digest,
-                                                             (const char **)FLATPAK_OCI_MANIFEST (versioned)->config.urls,
```

Comment:
as well as this

Reply:
This is actually okay. It only looks up the image config locally anyway so it makes no difference.